### ISSUE TITLE
[2DC] Starting replication after a backup restore should be efficient and safe

### ISSUE THREAD

**Issue by hectorgcr** [yugabyte/yugabyte-db#3319]
We need a way for the producer or the consumer to know what is a safe checkpoint to start replication from once a backup has been restored in the consumer. In other words, once the backup has been restored:
1. It's acceptable to receive a few unnecessary (duplicate) entries, but we should minimize those for efficiency.
2. There should never be any holes in the data. This implies that the necessary log files will be retained in the producer.

This will be achieved by introducing a new step which allows the user to create a checkpoint of the most recent op ids for all the tablets of a specific table. We call this step `bootstrapping` the producer. In practice this will be done by creating a new `yb-admin` command `bootstrap_cdc_producer`. When this command is executed:
1. The producer will create the `cdc_state` table if needed
2. For each table, create a new cdc stream id. 
3. For each tablet, we will insert a new record of the form `tablet_id, stream_id, <latest op id for that tablet>`,

To make log retention work based on the op id, the flag `enable_log_retention_by_op_idx` needs to be set to true. This is necessary so that the user has enough time to make a backup, restore the backup on the producer, and start replication.

After the producer is bootstrapped, a backup of the producer is taken. This backup is guaranteed to have all the entries with op ids created during the bootstrap process.

Then, the backup is restored on the consumer. Finally, unidirectional replication is started on the consumer (producer -> consumer) and the bootstrap ids are specified for each table. This will allow the producer to reuse the `cdc_state` entries created during the bootstrap step. When the first GetChanges request for a tablet is sent, it doesn't contain an op_id to start from. In this case, the producer will use the op id inserted during the bootstrap step.

In summary, these will be the steps:
1. Set flag `enable_log_retention_by_op_idx` to true in all the tservers
2. Bootstrap the producer by running the `yb-admin` command  `bootstrap_cdc_producer`
3. Create a backup in the producer
4. Restore the backup on the consumer
5. Start replication on the consumer by providing the bootstrap ids generated in step 2
